src/sync.cpp

```diff
@@ -140,16 +148,27 @@ namespace librealsense
         _name = create_composite_name(matchers, name);
     }
 
-    void composite_matcher::dispatch(frame_holder f, syncronization_environment env)
+    void composite_matcher::dispatch(frame_holder f, const syncronization_environment& env)
     {
+
         std::stringstream s;
-        s <<"DISPATCH "<<_name<<"--> "<< frame_to_string(f) <<"\n";
-        LOG_DEBUG(s.str());
+        s << "DISPATCH " << _name << "--> " << frame_to_string(f) << "\n";
+        log_if_enable(s.str(), env);
```

Comment:
I think it would be better as:

```
#define LOG_IF_ENABLE( OSTREAM, ENV ) \
	while( ENV.log ) \
	{ \
		LOG_DEBUG( OSTREAM ); \
		break; \
	}
```

Reply:
done